runtime/runtime/src/lib.rs

```diff
@@ -562,7 +565,8 @@ impl Runtime {
 
         // state_update might already have some updates so we need to make sure we commit it before
         // executing the actual receipt
-        state_update.commit();
+        state_update
+            .commit(StateChangeCause::ReceiptProcessing { receipt_hash: receipt.get_hash() });
```

Comment:
Should we split causes in this method into 3 different causes? It might be easier to parse later

Reply:
I would keep it simple for now and later revisit the cases. Let's merge it since we are blocking Flux.